src/Writer/QueryWriter.php

```diff
@@ -99,31 +99,60 @@ protected function writeFilter($source, Specification\FilterInterface $filter, C
             });
         }
 
-        if ($filter instanceof Specification\Filter\Like) {
+        if ($filter instanceof Specification\Filter\InjectionFilter) {
+            $expression = $filter->getFilter();
+            if ($expression instanceof Specification\Filter\Expression) {
```

Comment:
This check will be omitted when arrays will be inherited from expressions so injection filter will have explicitly passed expression sub-filter